Examples/Io/Csv/src/CsvSimHitReader.cpp

```diff
@@ -0,0 +1,93 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2019 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Io/Csv/CsvSimHitReader.hpp"
+
+#include "Acts/Utilities/Units.hpp"
+#include "ActsExamples/EventData/SimHit.hpp"
+#include "ActsExamples/Framework/WhiteBoard.hpp"
+#include "ActsExamples/Utilities/Paths.hpp"
+
+#include <dfe/dfe_io_dsv.hpp>
+
+#include "TrackMlData.hpp"
+
+ActsExamples::CsvSimHitReader::CsvSimHitReader(
+    const ActsExamples::CsvSimHitReader::Config& cfg, Acts::Logging::Level lvl)
+    : m_cfg(cfg),
+      // TODO check that all files (hits,cells,truth) exists
+      m_eventsRange(
+          determineEventFilesRange(cfg.inputDir, cfg.inputStem + ".csv")),
+      m_logger(Acts::getDefaultLogger("CsvSimHitReader", lvl)) {
+  if (m_cfg.inputStem.empty()) {
+    throw std::invalid_argument("Missing input filename stem");
+  }
+  if (m_cfg.outputSimulatedHits.empty()) {
+    throw std::invalid_argument("Missing simulated hits output collection");
+  }
+}
+
+std::string ActsExamples::CsvSimHitReader::CsvSimHitReader::name() const {
+  return "CsvSimHitReader";
+}
+
+std::pair<size_t, size_t> ActsExamples::CsvSimHitReader::availableEvents()
+    const {
+  return m_eventsRange;
+}
+
+ActsExamples::ProcessCode ActsExamples::CsvSimHitReader::read(
+    const ActsExamples::AlgorithmContext& ctx) {
+  auto path = perEventFilepath(m_cfg.inputDir, m_cfg.inputStem + ".csv",
+                               ctx.eventNumber);
+  // define all optional columns
+  std::vector<std::string> optionalColumns = {
+      "geometry_id", "tt",      "te",     "deltapx",
+      "deltapy",     "deltapz", "deltae", "index",
+  };
```

Comment:
These were only optional columns for the TrackMl dataset. Since the `SimHit{Reader,Writer}` does not need to be strictly compatible, we can deviate. I would suggest that we require all columns to be present. That simplifies our life since we know further down in the processing chain that all data is valid.